operator/v1alpha1/types_image_content_source_policy.go

```diff
@@ -0,0 +1,45 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ImageContentSourcePolicy holds cluster-wide information about how to handle registry mirror rules.  The canonical name is `cluster`
+type ImageContentSourcePolicy struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +required
+	Spec ImageContentSourcePolicySpec `json:"spec"`
+}
+
+type ImageContentSourcePolicySpec struct {
+	// DigestMappingRules holds the registry mirroring rules that are written to
+	// /etc/containers/registries.conf to be used by the runtime to determine which
+	// repo to fall back on, if the original repo cannot be accessed.
+	// We only support pulling by digests for mirrors, hence the name.
+	// Will also be used in cases such as offline installs/upgrades.
+	// +optional
+	DigestMappingRules DigestMappingRules `json:"digestMappingRules"`
```

Comment:
Other possible names `repositoryDigestMappings`

Reply:
`imageDigestMappings`. `imageDigestRetrieval`.

Reply:
(To me as a non-native speaker), all of these names seem to say that _digests_ are mapped, but they are not; digests are the only part of the image reference that don’t change.

`mirroredRepositories`, `mirrorSets`? OTOH to ensure that the digest restriction is not overlooked, that could end up with `mirorredRepositoriesInDigestReferences` or something similarly horrible.

Reply:
Changed it to `MirroredReposForDigests`

Reply:
`For digests` is probably not accurate - you're not mirroring for digests, your mirroring digests by mirrors.  We generally do not use short names inside api objects (reference instead of ref, for example).

Reply:
(I don’t really want to restart the naming discussion for this field, the current name is acceptable and I don’t think we’ll get a perfect one, just to minimize the use of this figure of speech:

We are never “mirroring digests”, we may not be mirroring [complete] repositories, we are mirroring OpenShift releases = sets of individual container images.